src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -2199,6 +2427,24 @@ private static void addLastSibling(DetailAstImpl self, DetailAstImpl sibling) {
         }
     }
 
+    /**
+     * Builds an imaginary DetailAstImpl with the given token details.
+     *
+     * @param tokenType the token type of this DetailAstImpl
+     * @param text the text of this DetailAstImpl
+     * @param lineNumber the line number of this DetailAstImpl
+     * @param columnNumber the column number of this DetailAstImpl
+     * @return imaginary DetailAstImpl from given details
+     */
+    private static DetailAstImpl buildImaginaryWithDetails(
+            int tokenType, String text, int lineNumber, int columnNumber) {
+        final DetailAstImpl imaginary = createImaginary(tokenType);
+        imaginary.setText(text);
```

Comment:
I understand why you called `createImaginary` but is it really needed in this case? `createImaginary` does `setText` but we overwrite it in the end. Seems better to just do the imaginary yourself here.

Maybe consider `createImaginary(int tokenType, String text) {`

Reply:
This is a good point, let me think on this and I will update.

Reply:
Done